compiler/src/dmd/dsymbol.d

```diff
@@ -1314,166 +1299,7 @@ public:
-                         * get actual overload roots which accessible via s and s2.
-                         */
-                        s = s.toAlias();
-                        s2 = s2.toAlias();
-                        /* If both s2 and s are overloadable (though we only
-                         * need to check s once)
-                         */
-
-                        auto so2 = s2.isOverloadSet();
-                        if ((so2 || s2.isOverloadable()) && (a || s.isOverloadable()))
-                        {
-                            if (symbolIsVisible(this, s2))
-                            {
-                                a = mergeOverloadSet(ident, a, s2);
-                            }
-                            if (!symbolIsVisible(this, s))
-                                s = s2;
-                            continue;
-                        }
-
-                        /* Two different overflow sets can have the same members
-                         * https://issues.dlang.org/show_bug.cgi?id=16709
-                         */
-                        auto so = s.isOverloadSet();
-                        if (so && so2)
-                        {
-                            if (so.a.length == so2.a.length)
-                            {
-                                foreach (j; 0 .. so.a.length)
-                                {
-                                    if (so.a[j] !is so2.a[j])
-                                        goto L1;
-                                }
-                                continue;  // the same
-                              L1:
-                                {   } // different
-                            }
-                        }
-
-                        if (flags & IgnoreAmbiguous) // if return NULL on ambiguity
-                            return null;
-
-                        /* If two imports from C import files, pick first one, as C has global name space
-                         */
-                        if (s.isCsymbol() && s2.isCsymbol())
-                            continue;
-
-                        if (!(flags & IgnoreErrors))
-                            ScopeDsymbol.multiplyDefined(loc, s, s2);
-                        break;
-                    }
-                }
-            }
-        }
-        if (s)
-        {
-            /* Build special symbol if we had multiple finds
-             */
-            if (a)
-            {
-                if (!s.isOverloadSet())
-                    a = mergeOverloadSet(ident, a, s);
-                s = a;
-            }
-            //printf("\tfound in imports %s.%s\n", toChars(), s.toChars());
-            return s;
-        }
-        //printf(" not found in imports\n");
-        return null;
-    }
-
-    extern (D) private OverloadSet mergeOverloadSet(Identifier ident, OverloadSet os, Dsymbol s)
+    extern (D) OverloadSet mergeOverloadSet(Identifier ident, OverloadSet os, Dsymbol s)
```

Comment:
So extern(D) methods need to be final to not affect the C++ layout? Makes sense.

Reply:
Hmmm... I'm sure I saw non final extern(D) methods in the codebase. I didn't really give much thought to this previously.

Reply:
They likely were in a `final class`.

Reply:
Sounds like the compiler should complain about this (non-`final` `extern(D)` methods on `extern(C++)` classes)

Reply:
Why?  They just occupy a vtable slot, but otherwise harmless.

It's `extern(C++)` classes with no virtual methods that are more likely to trip people up.